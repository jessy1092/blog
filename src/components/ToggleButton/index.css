
.button {
	background: #999;
	padding: 4px 10px;
	border-radius: 5px;
	border: none;
	color: #DDD;
}

.button-reverse {
	composes: button;
	background: #DDD;
	color: #999;
}
