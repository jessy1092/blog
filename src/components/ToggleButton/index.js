
import React from 'react';

import style from './index.css';

const ToggleButton = ({ toggle, handleClick }) => (
	<button className={toggle ? style.button : style.buttonReverse} onClick={handleClick}>
		{toggle ? 'Close' : 'Open'}
	</button>
);

export default ToggleButton;
