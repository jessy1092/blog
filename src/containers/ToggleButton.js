
import { contain } from 'react-container-helper';

import ToggleButton from '../components/ToggleButton';

const initState = () => ({
	toggle: true,
});

const mapStateToProps = ({ toggle }) => ({
	toggle,
});

const mapSetStateToProps = (setState, { toggle }, { onOpen, onClose }) => ({
	handleClick() {
		if (toggle) {
			onClose();
		} else {
			onOpen();
		}
		setState({ toggle: !toggle });
	},
});

export default contain(initState, mapStateToProps, mapSetStateToProps)(ToggleButton);
