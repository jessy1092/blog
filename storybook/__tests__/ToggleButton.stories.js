
import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import expect from 'expect';

import {
	storiesOf,
	action,
	describe,
	it,
	specs,
	snapshot,
} from '../facade';

import ToggleButton from '../../src/containers/ToggleButton';

const stories = storiesOf('ToggleButton', module);

stories.add('with default', () => {
	const toggleButton = (
		<ToggleButton
			onOpen={action('open')}
			onClose={action('close')}
		/>
	);

	specs(() => describe('ToggleButton', () => {
		it('should have default value: Close', () => {
			const component = mount(toggleButton);
			expect(component.text()).toContain('Close');
		});

		it('should become Open after click', () => {
			const component = mount(toggleButton);
			component.find('button').simulate('click');
			expect(component.text()).toContain('Open');
		});
	}));

	// Create custom snapshot testing
	snapshot('should become Open after click', (jestExpect) => {
		const component = renderer.create(toggleButton);

		let tree = component.toJSON();
		jestExpect(tree).toMatchSnapshot();

		// manually trigger the callback
		tree.props.onClick();
		// re-rendering would become Open
		tree = component.toJSON();
		jestExpect(tree).toMatchSnapshot();

		// manually trigger the callback
		tree.props.onClick();
		// re-rendering would become Close
		tree = component.toJSON();
		jestExpect(tree).toMatchSnapshot();
	});

	return toggleButton;
});
