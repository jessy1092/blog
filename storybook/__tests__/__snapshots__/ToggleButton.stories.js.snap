exports[`test should become Open after click 1`] = `
<button
  className={undefined}
  onClick={[Function]}>
  Close
</button>
`;

exports[`test should become Open after click 2`] = `
<button
  className={undefined}
  onClick={[Function]}>
  Open
</button>
`;

exports[`test should become Open after click 3`] = `
<button
  className={undefined}
  onClick={[Function]}>
  Close
</button>
`;

exports[`test with default 1`] = `
<button
  className={undefined}
  onClick={[Function]}>
  Close
</button>
`;
